test/ShellTest.php

```diff
@@ -384,6 +384,8 @@ public function testWriteStdout()
 
     public function testWriteStdoutWithoutNewline()
     {
+        $this->markTestSkipped('This test won\'t work on CI without overriding pipe detection');
```

Comment:
This test is always skipped, even not on the CI?

Reply:
It'll fail locally if you pipe your phpunit output, too :)

I didn't really want a mock or an override or anything for the piped output detection, but I think I'm going to need it.

Reply:
It feels like there might be one more configuration option missing. Some way to say "force this to act like output isn't piped", just like we have "force this to decorate output" and "force this to act like input is interactive".

I haven't decided yet whether that's exactly the same as "force this to decorate output", but it might be.